pkg/quota/v1/evaluator/core/pods.go

```diff
@@ -358,30 +359,11 @@ func PodUsageFunc(obj runtime.Object, clock clock.Clock) (corev1.ResourceList, e
 		return result, nil
 	}
 
-	requests := corev1.ResourceList{}
-	limits := corev1.ResourceList{}
-	// TODO: ideally, we have pod level requests and limits in the future.
-	for i := range pod.Spec.Containers {
-		containerRequests := pod.Spec.Containers[i].Resources.Requests
-		if feature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
-			cs, ok := podutil.GetContainerStatus(pod.Status.ContainerStatuses, pod.Spec.Containers[i].Name)
-			if ok && cs.ResourcesAllocated != nil {
-				containerRequests = quota.Max(containerRequests, cs.ResourcesAllocated)
```

Comment:
Just thinking out loud: If I'm not mistaken, quota evaluation happens much earlier in the workflow. Once the pod has been accepted by a node, a subsequent resize that leads to PodResizeStatusInfeasible does not alter the pod's quota usage. The pod will still be using whatever it was admitted at by the node. I do see your point that we may be overdoing it, but we made a conscious choice to conservatively use max here and err on the side of caution. It can always be relaxed later.